src/Init/Data/BitVec/Basic.lean

```diff
@@ -870,4 +870,14 @@ def clz (x : BitVec w) : BitVec w := clzAuxRec x (w - 1)
 /-- Count the number of trailing zeros. -/
 def ctz (x : BitVec w) : BitVec w := (x.reverse).clz
 
+/-- Count the number of bits with value `1` downward from the `pos`-th bit to the
+  `0`-th bit of `x`, storing the result in `acc`. -/
+def cpopNatRec (x : BitVec w) (pos acc : Nat) : Nat :=
+  match pos with
+  | 0 => acc
+  | n + 1 => x.cpopNatRec n (acc + (x.getLsbD n).toNat)
+
+/-- Count the number of bits with value `1` in `x`. -/
+def cpop (x : BitVec w) : BitVec w := BitVec.ofNat w (cpopNatRec x w 0)
```

Comment:
People are definitely going to look for `BitVec.popcount` and `BitVec.popcnt`, so the docstring should definitely the strings `popcount` and `popcnt`, and I think it would also be great if you would add `suggest_for` annotations for those two names (cf. #11554).